client/cmd.go

```diff
@@ -93,11 +93,6 @@ func ReadPersistentCommandFlags(clientCtx Context, flagSet *pflag.FlagSet) (Cont
 		clientCtx = clientCtx.WithOutputFormat(output)
 	}
 
-	if clientCtx.HomeDir == "" || flagSet.Changed(flags.FlagHome) {
```

Comment:
From my perspective, the code L96-L99 is redundant. We reuse the same code in [ReadHomeFlag](https://github.com/cosmos/cosmos-sdk/blob/cyberbono3/8529-add-client-config/client/cmd.go#L258). That's why I have decided to remove it.